components/new_channel_modal/new_channel_modal.tsx

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {useIntl} from 'react-intl';
+
+import classNames from 'classnames';
+
+import GenericModal from 'components/generic_modal';
+import Input from 'components/widgets/inputs/input/input';
+import PublicPrivateSelector from 'components/widgets/public-private-selector/public-private-selector';
+import URLInput from 'components/widgets/inputs/url_input/url_input';
+
+import {createChannel} from 'mattermost-redux/actions/channels';
+import Permissions from 'mattermost-redux/constants/permissions';
+import {switchToChannel} from 'actions/views/channel';
+import {closeModal} from 'actions/views/modals';
+import {DispatchFunc} from 'mattermost-redux/types/actions';
+import {ChannelType, Channel} from 'mattermost-redux/types/channels';
+import {ServerError} from 'mattermost-redux/types/errors';
+import {haveICurrentChannelPermission} from 'mattermost-redux/selectors/entities/roles';
+import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+
+import {GlobalState} from 'types/store';
+import Constants, {ModalIdentifiers} from 'utils/constants';
+import {cleanUpUrlable, validateChannelUrl, getSiteURL} from 'utils/url';
+import {localizeMessage} from 'utils/utils';
+
+import './new_channel_modal.scss';
+
+export function getChannelTypeFromPermissions(canCreatePublicChannel: boolean, canCreatePrivateChannel: boolean) {
+    let channelType = Constants.OPEN_CHANNEL;
```

Comment:
@julmondragon I really like the new modal. It would be great if we could get a setting to adjust this default to private. We run an instance where channels should generally be private, but with public selected as the default in the new channel modal, new channels are quite often made public by mistake. Having it default to private would reduce those mistakes.

Reply:
We only allow team admins and admins to create public channels in the first place, but those admins make mistakes too :sweat_smile: